src/Normalizer/EntityReferenceItemNormalizer.php

```diff
@@ -98,6 +103,16 @@ public function normalize($field_item, $format = NULL, array $context = []) {
         $arguments = isset($field_mappings['datatype_callback']['arguments']) ? $field_mappings['datatype_callback']['arguments'] : NULL;
         $values_clean['@value'] = call_user_func($callback, $target_entity, $arguments);
       }
+      $field = $field_item->getParent();
+      $field_context = $this->jsonldContextgenerator->getFieldsRdf(
+        $context['current_entity_rdf_mapping'],
+        $field->getName(),
+        $field->getFieldDefinition(),
+        $context['namespaces']
+      );
+      if (isset($field_context[$field_keys[0]])) {
+        $values_clean = $values_clean + $field_context[$field_keys[0]];
```

Comment:
If one of the vars at line 114 here is not an array it will fail via 
PHP Fatal error:  Unsupported operand types. This is what @dannylamb is seeing. Which means we need to make sure that everything is A) and array or cast. Not sure about the cast because it can lead to undesired results.